dWorldServer/WorldServer.cpp

```diff
@@ -234,6 +235,30 @@ int main(int argc, char** argv) {
 		Game::physicsWorld = &dpWorld::Instance(); //just in case some old code references it
 		dZoneManager::Instance()->Initialize(LWOZONEID(zoneID, instanceID, cloneID));
 		g_CloneID = cloneID;
+
+		// pre calculate the FDB checksum
+		if (bool(std::stoi(Game::config->GetValue("check_fdb")))) {
+				std::ifstream fileStream;
+				fileStream.open ("res/CDServer.fdb", std::ios::binary | std::ios::in);
+				const int bufferSize = 1024;
+				MD5* md5 = new MD5();
+				
+				while (!fileStream.eof()) {
+					char * fileStreamBuffer = new char[bufferSize];
```

Comment:
Allocate this on the stack, not on the heap. Currently this will cause a memory leak as it's not deleted. Faster and better to not allocate on the heap at all.

Reply:
I am swapping this out with a std::vector now anyway because this doesn't compile on MSVC

Reply:
You should be able to just make it a `char buffer[size]`

Reply:
> You should be able to just make it a `char buffer[size]`

Not if size isn’t a constexpr.

Reply:
Yeah, it only works on clang because of a compiler addon (iirc, ask avery about it), but you shouldn't be able to create a char array without the size being defined with a constexpr.

Reply:
> Yeah, it only works on clang because of a compiler addon (iirc, ask avery about it), but you shouldn't be able to create a char array without the size being defined with a constexpr.

Clang has a compiler extension that allows this. A compiler agnostic solution would be over-allocating in the first place or using a char vector.

Reply:
My latest commit makes it a char vector